packages/app-builder-lib/src/macPackager.ts

```diff
@@ -268,9 +273,17 @@ export class MacPackager extends PlatformPackager<MacConfiguration> {
         }
       }
 
+      if (qualifier === "-") {
+        identity = new Identity("-", undefined)
+      }
+
       if (!options.sign && identity == null) {
-        await reportError(isMas, certificateTypes, qualifier, keychainFile, this.forceCodeSigning)
-        return false
+        await reportError(isMas, certificateTypes, qualifier, keychainFile, this.forceCodeSigning, fallBackToAdhoc)
+        if (fallBackToAdhoc) {
+          identity = new Identity("-", undefined)
+        } else {
+          return false
+        }
       }
```

Comment:
This part seems a little messy as I'm generally averse to nesting logic when not necessary and I'm seeing `reportError` still having a `throw new Error()`, so the `fallBackToAdhoc` would never be executed.  Maybe this would work? 
```
const noIdentity = !options.sign && identity == null
if (qualifier === "-" || (noIdentity && fallBackToAdhoc)) {
    identity = new Identity("-", undefined)
} else if (noIdentity) {
    await reportError(isMas, certificateTypes, qualifier, keychainFile, this.forceCodeSigning)
    return false
}
```

Reply:
I still think `reportError` should only be used for throwing if no certificates are valid, whereas this `noIdentity && fallBackToAdhoc` could just include a `log.info` logic flow saying that it was selected as a fallback 
```
const noIdentity = !options.sign && identity == null
if (qualifier === "-") {
    identity = new Identity("-", undefined)
} else if (noIdentity && fallBackToAdhoc) {
    log.info("falling back to ad-hoc signature for macOS application code signing")
    identity = new Identity("-", undefined)
} else if (noIdentity) {
    await reportError(isMas, certificateTypes, qualifier, keychainFile, this.forceCodeSigning)
    return false
}
```
Thoughts?

Reply:
I've applied those changes. I also adjusted `fallBackToAdhoc` to avoid falling back if force code signing is enabled, since allowing ad-hoc signing with that setting enabled would probably be unexpected for most people. I also set the log message to the warn level.

Reply:
Nice catch!